File: app/src/test/java/com/tronography/pixelweather/DateFormatterTest.java
package com.tronography.pixelweather;

import android.text.format.DateUtils;

import com.tronography.pixelweather.utils.DateFormatter;

import org.junit.Assert;
import org.junit.Test;


public class DateFormatterTest {

    @Test
    public void dateFormatterTest(){
        long dateToFormat = 1508095680;
        String expectedResult = "SUN\n10/15 3pm";

        Assert.assertEquals(expectedResult, DateFormatter.INSTANCE.forecastDateFormatter(dateToFormat));
    }
}

File: app/src/test/java/com/tronography/pixelweather/DateFormatterTest.kt
package com.tronography.pixelweather

import android.text.format.DateUtils

import com.tronography.pixelweather.utils.DateFormatter

import org.junit.Assert
import org.junit.Test


class DateFormatterTest {

    @Test
    fun dateFormatterTest() {
        val dateToFormat: Long = 1508095680
        val expectedResult = "SUN\n10/15 3pm"

        Assert.assertEquals(expectedResult, DateFormatter.forecastDateFormatter(dateToFormat))
    }

    @Test
    fun hourlyFormatTest() {
        val dateToFormat: Long = 1508095680
        val expectedResult = "3pm"

        Assert.assertEquals(expectedResult, DateFormatter.hourlyFormat(dateToFormat))
    }

    @Test
    fun dayFormatTest() {
        val dateToFormat: Long = 1508095680
        val expectedResult = "SUN"

        Assert.assertEquals(expectedResult, DateFormatter.dayFormat(dateToFormat))
    }
}
